app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -270,15 +270,30 @@ def sanitize_inherited_vm_attributes(args)
       end
       args.delete(:cores) if (args[:cores].blank? && cores) || (args[:cores].to_i == cores)
       args.delete(:memory) if (args[:memory].blank? && memory) || (args[:memory].to_i == memory)
+
+      template_volumes = template.volumes
+
+      disks = []
+
+      args[:volumes_attributes].values.select { |x| x[:id].present? && x[:preallocate] == '0' }.each do |volume|
```

Comment:
This needs to be refactored and combined with `preallocate_disks` but opening now for testing of the `fog-ovirt` PR and I'll continue working on it.